models/clm/readme.rst

```diff
@@ -1,40 +1,195 @@
+Anything with snow layers, for example. CLM has the *SNLSNO* varible to indicate
+which snow layers are active. The *unused* layers may not have the *_FillValue*
+value, but can have 'indeteminate' values. The :doc:`clm_to_dart` 
+must be run to convert these indeterminate values to *_FillValue* to be 
+interpreted correctly by DART.  After the assimilation is complete, the 
+:doc:`dart_to_clm` must be called to replace the *_FillValue* with whatever
+is originally in that slot. Extra care is taken to handle the special case
+of a *trace* of snow. See the *Discussion of Indeterminate Values* 
+section of :doc:`clm_to_dart` for more details.
+
+
+Model Interpolate - The Forward Operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since the subgridscale components of CLM have no explicit location associated 
+with them, the location of every component in the gridcell is the same as the 
+gridcell itself. The DART forward operators fundamentally rely on 
+interpolating the model state to some arbitrary location. At present, the best we
+can do is to create an area-weighted average of all pieces in the gridcell.
+This is clearly sub-optimal. A nice project would be to use some sort of lookup
+table for the observation location to determine the dominant PFT or whatever at
+that location and then just average up all the similar PFTs in the gridcell.
+The forward operator would be more accurate and might have a discernable impact
+on the regression relationship (i.e. ensemble covariance) with the variables
+in the DART state vector.
+
+The *model_interpolate* function in DART achieves efficiency by interpolating
+all the ensemble members at the same time. This gives rise to some challenging
+problems when interpolating values for variables with dynamic active layers.
+Some ensemble members may only have 2 active snow layers, some may have 3 ...
+this is an untenable situation when asked for the snow temperature or water 
+content in layer 3, for example. Consequently - *model_interpolate* will fail
+and return an error code - the forward operator will fail - and the observation
+is rejected and the DART QC is marked as such. Be aware.
+
+Localization
+~~~~~~~~~~~~
+
+Localization is the term used to restrict the portion of the state to the portion
+believed to be related to the observation. Most often, this is a spatial argument
+but it does not need to be restricted to that. In some way, even the selection
+of the CLM variables to include in the DART state is a de-facto localization.
+Since CLM has such a rich description of land unit types: urban columns, glaciers, 
+lakes, etc. it is also possible (and probably desirable) to explicitly declare
+some columns and/or PFTs to be unaffected by the assimilation - i.e., we are
+going to declare that soil moisture observations should not impact urban columns
+or deep lakes or ... The **get_close_state()** function employs a routine to
```

Comment:
Instead of 'we are going' -- which sounds like a work in progress, suggest 'we declare'.   This is the current way we localize based on land cover type.

Reply:
I changed the wording to be less repetetive, actually.